### PR TITLE
front: rollingstockeditor: add create form metadata

### LINKED ISSUE #4014: Add "create button" with empty form in RS editor
## Description

The rolling stock editor needs to allow user to create new rolling stocks from scratch

## Goal

Allow studies with new rolling stocks, for example to compute their running time on a given path

## Acceptance criteria

- [ ] There must be a "Add" button (or equivalent) at the bottom of the rolling stock list in the editor
- [ ] Clicking this button should open an empty creation form

### LINKED ISSUE #4343: Display locked rolling stocks in the interface

### PR DESCRIPTION
**Pull request by Yohh**
close #4014 
close #4343

### CODE REVIEW

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 1]
```diff
@@ -53,33 +64,38 @@ const RollingStockEditorInput = ({
...
+        className={`form-control-sm bg-white px-2 ${
+          !position && label.includes('rolling_resistance') ? 'col-3' : 'col-4'
+        }`}
         {...register(label)}
       />
-      {!position && (
-        <span className="col-1">
-          {label.substring(label.length - 1, label.length).toLocaleLowerCase()}:{' '}
-        </span>
+      {!position && label.includes('rolling_resistance') && (
```
Comment (alexandredamiron) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: It won't work anymore the day the labels changes. Use a boolean: isRollingResistance in your RollingStockEditorElementsProps for instance

File: front/src/styles/scss/applications/rollingStockEditor/_rollingStockForm.scss [hunk 2]
```diff
@@ -1,20 +1,75 @@
...
+            }
+          }
+        }
+      }
+    }
+
+    width: 100%;
     align-self: center;
+    @media screen and (min-width: 767px) {
+    }
```
Comment (alexandredamiron) [inline front/src/styles/scss/applications/rollingStockEditor/_rollingStockForm.scss]: An emptu query ? Why

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 3]
```diff
@@ -114,44 +129,91 @@ const RollingStockEditorSelect = ({
...
+          property.match(/detail|family|grouping|number|reference|series|subseries|type|unit/) &&
+          (position ? index < 5 : index >= 5)
+      )
+      .map((property) => (
+        <div key={property}>
+          <RollingStockEditorInput
+            label={property}
+            register={registerForm}
+            position={position}
+            error={errors[property] as unknown as string}
```
Comment (alexandredamiron) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: We are not superfan of this typecasting

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 4]
```diff
@@ -114,44 +129,91 @@ const RollingStockEditorSelect = ({
...
   } as UseFormProps<RollingStockParametersValues>);
 
-  // in the left side of the form, we display all the information except the rollingStock resistance params,
+  // in the top of the form we display all the metadatas
+  // the array is split in two to display the metadatas in two columns
+  const metadataForm = (position: boolean, registerForm: UseFormRegister<FieldValues>) =>
+    Object.keys(schema.properties)
+      .filter(
+        (property, index) =>
+          property.match(/detail|family|grouping|number|reference|series|subseries|type|unit/) &&
```
Comment (alexandredamiron) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: use of regex are hard to maintain. Can you set this kind of behavior through the schema ?

Comment (alexandredamiron) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: or at least your regex should be a param

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 3 above)
Comment (alexandredamiron) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: just errors[property] as string should work

**Comment by alexandredamiron** [comment]
Capture d’écran 2023-06-28 à 14 57 20

no bottom grey line to enclose the card

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 5]
```diff
@@ -114,44 +129,91 @@ const RollingStockEditorSelect = ({
...
+  // in the left side of the form, under the metadatas, we display all the information except the rollingStock resistance params,
   // which are displayed in the right part of the form
   const parameterForm = (position: boolean, registerForm: UseFormRegister<FieldValues>) =>
     Object.keys(schema.properties)
+      .filter(
+        (property) =>
+          property.match(
+            /startup_time|startup_acceleration|comfort_acceleration|gamma_value|inertia_coefficient|loading_gauge|rolling_resistance|length|mass|max_speed/
+          ) && !property.match(/electrical_power_startup_time/)
+      )
```
Comment (alexandredamiron) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: same. Us of hard-coded regex inline is hard to maintain. Use the typescript RegExp interface and at least set a param prop or a const

File: front/src/styles/scss/applications/rollingStockEditor/_rollingStockForm.scss (hunk 2 above)
Comment (Yohh) [reply front/src/styles/scss/applications/rollingStockEditor/_rollingStockForm.scss]: oups!

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 3 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: ok

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 1 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: ok

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 5 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: ok

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 3 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: done

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 4 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: done

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 5 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: done

File: front/src/styles/scss/applications/rollingStockEditor/_rollingStockForm.scss (hunk 2 above)
Comment (Yohh) [reply front/src/styles/scss/applications/rollingStockEditor/_rollingStockForm.scss]: done

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 1 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: done

**Review by alexandredamiron — APPROVED** [review]

**Pull request merged**